### ISSUE TITLE
Wiremock's default gzip encoder adds ~28 bytes to the "Content-Lenght" header

### ISSUE THREAD

**Issue by AndriiYaremchuk** [wiremock/wiremock#1771]
**Wiremock version**: 2.32.0
 **Wiremock config**: 
```
WireMockServer wireMockServer = WireMockProvider.get().startWireMockServer(mockName,
            options().port(mockHttpPort)
                .httpsPort(Options.DYNAMIC_PORT)
                .keystorePath(keyStorePath)
                .bindAddress(bindAddress)
                .extensions(BasicResponseTransformer.class,
                    BasicRequestResponseTransformer.class,
                    DecodeRequestResponseTransformer.class)
                .withRootDirectory(Paths.get("build/mock-servers/" + mockName).toAbsolutePath().toString()));
```
 **Request headers**
```
  "content-type" : "application/json",
      "app_brand" : "***",
      "x-api-lang" : "en",
      "hardware_id" : "***",
      "accept-encoding" : "gzip",
      "accept" : "image/h264, image/jpeg",
      "user-agent" :  "***"
```
 **Case details**:
  When we proxy our requests through the Wiremock proxy with headers ` "accept" : "image/h264, image/jpeg", "accept-encoding" : "gzip"` we receive the wrong "Content-Lenght" header response. It looks like Wiremock adds ~ 28 bytes to the header on decoding/encoding image/h264 while proxying. No transformations are done during this request.
 
 **Steps to reproduce:**
- Send any request with `"accept-encoding" : "gzip"` and  default `gzipDisabled` option (which is set to `false` i.e gzip encoding/decoding is enabled by default) to the server which should return any media file encoded in h/264 ByteStream through the Wiremock proxy requests option;
- Verify ByteStream length and "Content-Lenght" header values. They should be the same.
In our case app receives 
`Failure OtherException(throwable=java.io.IOException: Content-Length (23612) and stream length (23584) disagree)`
_Note:_ Our case passes when we set `gzipDisabled` to true. This is not the issue solver as we do need gzip encoding/decoding on some requests.

**Comment by AndriiYaremchuk** [comment]
Any chance for this to be looked at?

**Comment by tomakehurst** [comment]
If you hit the API you're proxying to directly do you get stream length = content-length = 23612 bytes?

If so, the issue is possibly that the content-length header is being passed through from the target's response, rather than calculated by WireMock, and the gzip settings and thus resulting size differ.

**Comment by AndriiYaremchuk** [comment]
No, if we hit the API directly (Postman call) we receive 23584 Content-length header value. If we set `gzipDisabled` to true we also receive 23584 content length.

**Comment by tomakehurst** [comment]
Please can you share all the steps necessary to reproduce this?

A working test case would be ideal.
